surface/include/pcl/surface/impl/texture_mapping.hpp

```diff
@@ -576,9 +577,16 @@ pcl::TextureMapping<PointInT>::sortFacesByCamera (pcl::TextureMesh &tex_mesh, pc
 
     // find occlusions on transformed cloud
     std::vector<int> visible, occluded;
+    std::unordered_set<int> occluded_set;
     removeOccludedPoints (transformed_cloud, filtered_cloud, octree_voxel_size, visible, occluded);
     visible_pts = *filtered_cloud;
 
+    // pushing occluded idxs into a set for faster lookup
+    for (int occluded_idx = 0; occluded_idx < (int)occluded.size(); ++occluded_idx)
+    {
+      occluded_set.insert(occluded[occluded_idx]);
+    }
```

Comment:
Can't this loop be replaced with `occluded_set.insert(occluded.begin(), occluded.end());`? Arguably you should even be able to construct directly the unordered set from a pair of iterators. This could avoid some operations.

Reply:
Yup, you're totally right. Just updated accordingly.